### PR TITLE
Remove unused results field from pr and tr specs

### LINKED ISSUE #1243: Remove unused Results field
# Expected Behavior

Our API should only contain fields we actually use.

# Actual Behavior

[The results field](https://github.com/tektoncd/pipeline/blob/dbd2e69ca91af93943e16d4beb9e12d72f1c60f7/pkg/apis/pipeline/v1alpha1/pipelinerun_types.go#L57) was part of our very first design and was imagine to contain info like links to logs and test results, but we have since not done anything with it. When we choose to revisit these topics we can add in fields as needed safely, but we should remove this field for now b/c it does nothing at all and the longer we leave it in, the greater the chances that someone will come to rely on it for some reason.

### PR DESCRIPTION
**Pull request by dibyom**
# Changes

Fixes #1243 
The `results` field is unused, was deprecated in 0.7, and was
scheduled to be removed in the 0.8 release

Co-authored-by: Daniel Helfand <dhelfand@redhat.com>
Signed-off-by: Dibyo Mukherjee <dibyo@google.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

# Release Notes

```
The `results` field has been removed from the TaskRun and PipelineRun spec
```

### CODE REVIEW

**Comment by dibyom** [comment]
Danierl has previously committed to pipelines so I'm guessing its the co-authorship that messed up the CLA

File: third_party/VENDOR-LICENSE [hunk 1]
```diff
@@ -1802,13 +1802,26 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 Import: github.com/tektoncd/pipeline/vendor/github.com/gobuffalo/envy
 
 The MIT License (MIT)
-Copyright (c) 2018 Mark Bates
 
-Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+Copyright (c) 2019 Mark Bates
```
Comment (dlorenc) [inline third_party/VENDOR-LICENSE]: Were these vendor changes intentional?

Comment (dibyom) [reply third_party/VENDOR-LICENSE]: I ran `./hack/update-codegen.sh` which updated this file

Comment (dlorenc) [reply third_party/VENDOR-LICENSE]: Ah strange, looks innocuous anyway...

**Comment by danielhelfand** [comment]
@dibyom Thanks for the co-authorship!

**Pull request merged**